### PR TITLE
Add guide on configuring gRPC routes

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Fixes: #1894

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
TODO: will convert to numbered links once we're done with reviews/edits

File: site/content/guides/grpc.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    app.kubernetes.io/name: grpc-echo
+  ports:
+  - port: 9000
+    protocol: TCP
+    targetPort: grpc
+```
+
+## HTTPProxy Configuration
+
+Configuring proxying to a gRPC service with HTTPProxy is as simple as specifying the protocol Envoy uses with the upstream application via the `spec.routes[].services[].protocol` field.
```
Comment (skriss) [inline site/content/guides/grpc.md:59]: Do we also want to suggest setting the route response timeout to a large value (ref. https://github.com/projectcontour/contour/issues/3692#issuecomment-1004192741)?

**Review by skriss — COMMENTED** [review]
Looks great overall 👍

File: site/content/guides/grpc.md (hunk 1 above)
Comment (sunjayBhatia) [reply site/content/guides/grpc.md:59]: yee definitely, good guideline to add for streaming RPCs

**Review by skriss — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
codegen failing should be fixed by https://github.com/projectcontour/contour/pull/4734

**Pull request merged**